### ISSUE TITLE
Cotangent Weights Don't Seem To Be Working Properly

### ISSUE THREAD

**Issue by ctralie** [ctralie/pyhks#1]

**Comment by yitao416** [comment]
Hello,

I think there are some issues from line 73 to 78. 

`    
 
    #Scale each row by the incident areas
    Areas = sparse.coo_matrix((VA, (IA, IA)), shape=(N, N)).tocsr()
    Areas = Areas.todia().data.flatten()
    Counts = sparse.coo_matrix((VC, (IA, IA)), shape=(N, N)).tocsr()
    Counts = Counts.todia().data.flatten()
    RowScale = sparse.dia_matrix((3*Counts/Areas, 0), L.shape)
    L = L.T.dot(RowScale).T
`

Because without the rowScale section, the LapCotWeight can work well with the getHKS function.

**Comment by ctralie** [comment]
Good catch, thank you!  This is definitely where the problem is.  I should
have come up with an area normalization based on the triangles.  For now,
you can indeed just comment these lines out, and the result should still be
visually better than with umbrella weights.  But the most correct way would
be to have the area normalization.

**Comment by ZhichaoWang970201** [comment]
Hey Ctralie,

I am new to HKS. Would you mind providing some reference papers on area normalization?

Thank you.

**Comment by ctralie** [comment]
Sure, here are some notes:
http://www.cs.cmu.edu/~kmcrane/Projects/Other/nDCotanFormula.pdf
https://cs.nyu.edu/~panozzo/gp/CotanLaplacian.pdf

But you may also be interested in the cotan_laplacian implementation here:
https://github.com/nmwsharp/potpourri3d
Best,
Chris

**Comment by ZhichaoWang970201** [comment]
![image](https://user-images.githubusercontent.com/50489235/119430866-829d2e00-bcdf-11eb-8b75-68810898b0b4.png)
In line 68 to line 74, should there be a one half for each element in the matrix L?

**Comment by ZhichaoWang970201** [comment]
![image](https://user-images.githubusercontent.com/50489235/119430973-b710ea00-bcdf-11eb-94c0-8b15549ad818.png)
In line 68 to 72, should there be a one half for each element in matrix L?

**Comment by ctralie** [comment]
Yes, but more importantly, the voronoi area is needed in the denominator
